### PR TITLE
feat: layout job detail

### PR DESCRIPTION
**Pull request by maulanasdqn**
**Todo**
- [x] Layout Job Detail https://app.clickup.com/t/86ep5tccm

**Screenshot / Image**
image

image

**Note**
i just install two package and need to run
- ts-pattern

`npm install`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "midtrans-client": "^1.3.1",
     "next": "^14.1.3",
     "next-auth": "^5.0.0-beta.15",
+    "next-usequerystate": "^1.17.1",
```
Comment (dwicaksono) [inline package.json]: Ini buat apa ya mas ? Handle query params kah? @maulanasdqn 
cc: @RikoOrlando

**Comment by dwicaksono** [comment]
@maulanasdqn boleh masukin url tiketya mas di setiap task check list nya ?

File: app/(dashboard)/(routes)/job/create/_modules/job-description.tsx
Comment (dwicaksono) [inline app/(dashboard)/(routes)/job/create/_modules/job-description.tsx:1]: folder _modules ini untuk apa ya ?
IMO, organizing all routes within the "app" folder would better focus on route management. If you want to create a component, simply move it to the "components" folder. You can create a new folder or a nested folder based on the name of the app's router.

File: components/ui/steper.tsx
Comment (dwicaksono) [inline components/ui/steper.tsx:1]: Is this component from the Shadcn package? Let's reserve the "ui" folder specifically for Shadcn components.
biar tau aja yang mana dari package shadcn componentnya dan yang mana di buat sendiri.

File: app/(dashboard)/(routes)/job/create/_modules/main.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { UploadCvModule } from './upload-cv';
+import { JobDescriptionModule } from './job-description';
+import { parseAsString, useQueryState } from 'next-usequerystate';
+import { match } from 'ts-pattern';
+import { Stepper } from '@/components/ui/steper';
+
+export const CreateJobModule: FC = (): ReactElement => {
+  const [step] = useQueryState('step', parseAsString.withDefault('job-detail'));
+
+  const JobModule = match(step)
```
Comment (dwicaksono) [inline app/(dashboard)/(routes)/job/create/_modules/main.tsx]: I'm not familiar with this package. Could you please explain its main functions and purposes to me?.
This seems to be conditional reading based on query parameters, is that correct?

**Comment by maulanasdqn** [comment]
siap mas @dwicaksono

**Comment by maulanasdqn** [comment]
ini gak ada opsi draft nya euy mau di draft teh

**Comment by dwicaksono** [comment]
Saat ini blm ada mas opsi draft @maulanasdqn

File: app/(dashboard)/(routes)/job/create/_modules/main.tsx (hunk 2 above)
Comment (maulanasdqn) [reply app/(dashboard)/(routes)/job/create/_modules/main.tsx]: i will remove this package

File: components/ui/steper.tsx
Comment (maulanasdqn) [reply components/ui/steper.tsx:1]: okey pindah ke share ya

File: app/(dashboard)/(routes)/job/create/_modules/job-description.tsx
Comment (maulanasdqn) [reply app/(dashboard)/(routes)/job/create/_modules/job-description.tsx:1]: already removed

File: package.json (hunk 1 above)
Comment (maulanasdqn) [reply package.json]: yep something like that

Comment (dwicaksono) [reply package.json]: possible ga kalau ga pake library ? @maulanasdqn 
cc : wdyt ? @RikoOrlando

File: app/(dashboard)/layout.tsx [hunk 3]
```diff
@@ -25,7 +25,7 @@ const DashboardLayout = async ({ children }: DashboardLayoutProps) => {
 
       </div> */}
```
Comment (dwicaksono) [inline app/(dashboard)/layout.tsx:26]: ini delete aja kayaknya @maulanasdqn

File: package.json (hunk 1 above)
Comment (maulanasdqn) [reply package.json]: aman sih gak kepake juga

**Comment by maulanasdqn** [comment]
updated

File: components/share/steper.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { FC, ReactElement } from 'react';
+
+type TStepper = {
+  step: string;
+  items: Array<string>;
+};
+
+export const Stepper: FC<TStepper> = (props): ReactElement => {
```
Comment (dwicaksono) [inline components/share/steper.tsx]: ini stepper component step A to B and can be more then ones ?
i think you can check revamp ui page figma to make sure the hires of design. mas @maulanasdqn 
and i think this component just on create page job.

File: app/(dashboard)/layout.tsx [hunk 5]
```diff
@@ -25,7 +25,7 @@ const DashboardLayout = async ({ children }: DashboardLayoutProps) => {
...
       <Sidebar />
-      <main className="w-full overflow-clip">
+      <main className="flex h-full w-full flex-col gap-y-3 overflow-clip">
```
Comment (dwicaksono) [inline app/(dashboard)/layout.tsx]: https://www.w3schools.com/tags/tag_main.asp#:~:text=Definition%20and%20Usage,site%20logos%2C%20and%20search%20forms.

File: app/(dashboard)/(routes)/job/[id]/template.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    {
+      text: 'Contacted',
+      link: '#',
+      icon: <Camera className="h-5 w-5" />,
+      notificationCount: 24,
+    },
+  ];
+  return (
+    <main className="w-full px-3">
```
Comment (dwicaksono) [inline app/(dashboard)/(routes)/job/[id]/template.tsx]: https://www.w3schools.com/tags/tag_main.asp#:~:text=Definition%20and%20Usage,site%20logos%2C%20and%20search%20forms.

File: app/(dashboard)/layout.tsx (hunk 3 above)
Comment (maulanasdqn) [reply app/(dashboard)/layout.tsx:26]: okey

File: app/(dashboard)/(routes)/job/[id]/template.tsx (hunk 6 above)
Comment (maulanasdqn) [reply app/(dashboard)/(routes)/job/[id]/template.tsx]: better pake section?

File: components/share/steper.tsx (hunk 4 above)
Comment (maulanasdqn) [reply components/share/steper.tsx]: i will remove this

File: app/(dashboard)/layout.tsx (hunk 5 above)
Comment (maulanasdqn) [reply app/(dashboard)/layout.tsx]: ini dashboard layout beneran mau di delete?

Comment (dwicaksono) [reply app/(dashboard)/layout.tsx]: enggak di delete, ini parent kan ya ? "`<main>{children}</main>`"
ask: kalau di didalamnya ada "`<main>` "lagi masih simantic ga ? kalau mau tetap main simantic tag.

note: yang di delete yang code di comment aja, mas @maulanasdqn

Comment (maulanasdqn) [reply app/(dashboard)/layout.tsx]: ganti ke tag apa mas? soalnya sebelum ku edit dia udh tag main dari sananya

**Comment by maulanasdqn** [comment]
updated mas @dwicaksono

File: app/(dashboard)/layout.tsx (hunk 5 above)
Comment (dwicaksono) [reply app/(dashboard)/layout.tsx]: IMO kalau parent terluar udah ada `<main>` di childnya ga perlu ada main tag lagi, cmiiw.
kalau mas @maulanasdqn punya ide yang lebih bagus silahkan.

**Pull request merged**